### PR TITLE
Issue #32 file to config

### PR DESCRIPTION
**Pull request by andrewmadden**
* Change the computation of config depending on require setting.
* Use the uploaded file to create the config.
* Unit tests changed to account for different require settings.
* Unit tests added for upload config.

### CODE REVIEW

File: classes/quiz_settings.php [hunk 1]
```diff
@@ -275,19 +275,54 @@ private function compute_config_key() {
...
+     *
+     * @throws \CFPropertyList\IOException
+     * @throws \CFPropertyList\PListException
+     * @throws \DOMException
+     * @throws \coding_exception
+     */
+    private function process_seb_config_file() {
+        $cm = get_coursemodule_from_instance('quiz', $this->get('quizid'));
+
+        $file = settings_provider::get_module_context_sebconfig_file($cm->id);
```
Comment (dmitriim) [inline classes/quiz_settings.php:311]: we can probably get rid of using settings_provider here, if we would store id of the file as sebconfigfile. Thoughts?

File: classes/settings_provider.php [hunk 2]
```diff
@@ -367,6 +367,26 @@ public static function get_current_user_draft_file(string $itemid) : ?stored_fil
...
+     * @param string $itemid Item ID of the file.
+     * @return stored_file|null Returns null if no file is found.
+     *
+     * @throws \coding_exception
+     */
+    public static function get_module_context_sebconfig_file(string $cmid) : ?stored_file { // @codingStandardsIgnoreLine
+        $fs = new \file_storage();
+        $context = context_module::instance($cmid);
+
+        if (!$files = $fs->get_area_files($context->id, 'quizaccess_seb', 'filemanager_sebconfigfile', $cmid,
```
Comment (dmitriim) [inline classes/settings_provider.php:382]: we can abstract this in the function as we use the same code in other place.

File: classes/quiz_settings.php (hunk 1 above)
Comment (dmitriim) [reply classes/quiz_settings.php:311]: but then it will add complexity on backup and restore...

Comment (nhoobin) [reply classes/quiz_settings.php:311]: Don't we still need the file somehow? Even if the file is referenced using the $cmid as the file $itemid, it must still be obtained from the file_storage api. This function returns the file itself to parse it and make changes to the quiz_settings plist.

**Pull request merged**